src/dev/no_pkg_semver_ranges/index.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { readFileSync, writeFileSync } from 'fs';
+import { resolve } from 'path';
+import { REPO_ROOT } from '@kbn/repo-info';
+
+const PKG_JSON_PATH = resolve(REPO_ROOT, 'package.json');
+const FIELDS_TO_CHECK = ['dependencies', 'devDependencies', 'resolutions', 'engines'] as const;
+const pkg = JSON.parse(readFileSync(PKG_JSON_PATH, 'utf-8'));
+
+type FieldName = (typeof FIELDS_TO_CHECK)[number];
+
+let totalFixes = 0;
+const fixesPerField: Partial<Record<FieldName, number>> = {};
+
+for (const field of FIELDS_TO_CHECK) {
+  const deps = pkg[field];
+  if (!deps || typeof deps !== 'object') continue;
+
+  for (const [name, version] of Object.entries(deps)) {
+    if (typeof version !== 'string') continue;
+
+    if (version.startsWith('^') || version.startsWith('~')) {
+      deps[name] = version.slice(1);
```

Comment:
I'm not sure this is the correct fix.
`^1.2.4` might refer to anything within the `1.x.x` scope, so correcting it to `1.2.4` might be a downgrade. I think a more appropriate fix here could be to look up the resolved version of that package, either from `node_modules/*/package.json` or from `yarn.lock`.

Reply:
This is meant to be a quick check for the new added ones from now on so what we change at this PR will be the baseline going forward. In any case its your PR and what you mentioned is a richer solution.